src/rp2/abstract_accounting_method.py

```diff
@@ -41,11 +47,17 @@ def __init__(
     ) -> None:
         self.__accounting_method: AbstractAccountingMethod = accounting_method
         self.__acquired_lot_list = acquired_lot_list
+        self.__acquired_lot_heap: List[Tuple[Union[float, RP2Decimal], InTransaction]] = []
         self.__acquired_lot_2_partial_amount = acquired_lot_2_partial_amount
         self.__to_index = 0
         self.__from_index = 0
 
     def set_to_index(self, to_index: int) -> None:
+        # Control how far to advance the iterator, caller is responsible for updating
+        if self.__accounting_method.use_heap():
+            for i in range(self.__to_index, to_index + 1):
```

Comment:
Not sure I'm getting this loop: could you clarify the intent?

Reply:
So it's getting called here within the accounting_engine: https://github.com/eprbell/rp2/blob/main/src/rp2/accounting_engine.py#L212

The lots under consideration are determined by the timestamp of the `taxable_event` here: https://github.com/eprbell/rp2/blob/main/src/rp2/accounting_engine.py#L202-L204

This loop is simply advancing the heap to the timestamp of the `taxable_event`.